pkgs/servers/web-apps/mealie/default.nix

```
@@ -0,0 +1,123 @@
+{ python310, python310Packages, fetchFromGitHub, mkYarnPackage, lib, ...}: let
+  version = "v1.0.0-RC2";
+  src = fetchFromGitHub {
+    owner = "mealie-recipes";
+    repo = "mealie";
+    rev = version;
+    sha256 = "sha256-/sht8s0Nap6TdYxAqamKj/HGGV21/8eYCuYzpWXRJCE=";
+  };
+
+  frontend = (mkYarnPackage {
+    name = "mealie-frontend";
+    src = "${src}/frontend";
+    packageJSON = "${src}/frontend/package.json";
+    yarnLock = "${src}/frontend/yarn.lock";
+  }).overrideAttrs (old:
+    {
+      doDist = false;
+
+      buildPhase = ''
+        runHook preBuild
+        shopt -s dotglob
+
+        rm deps/${old.pname}/node_modules
+        mkdir deps/${old.pname}/node_modules
+        pushd deps/${old.pname}/node_modules
+        ln -s ../../../node_modules/* .
+        popd
+        export NUXT_TELEMETRY_DISABLED=1
+        yarn --offline --non-interactive --production=false build
+        yarn --offline --non-interactive --produciton=false generate
+        runHook postBuild
+      '';
+
+      installPhase = ''
+        runHook preInstall
+        mv deps/${old.pname}/dist $out
+        runHook postInstall
+      '';
+    });
```

At the very least, you shouldn't need `overrideAttrs` (my indentation might be wrong)

```suggestion
  frontend = mkYarnPackage {
    name = "mealie-frontend";
    src = "${src}/frontend";

    packageJSON = "${src}/frontend/package.json";
    yarnLock = "${src}/frontend/yarn.lock";

    doDist = false;

    buildPhase = ''
      runHook preBuild
      shopt -s dotglob

      rm deps/${old.pname}/node_modules
      mkdir deps/${old.pname}/node_modules
      pushd deps/${old.pname}/node_modules
      ln -s ../../../node_modules/* .
      popd
      export NUXT_TELEMETRY_DISABLED=1
      yarn --offline --non-interactive --production=false build
      yarn --offline --non-interactive --produciton=false generate

      runHook postBuild
    '';

    installPhase = ''
      runHook preInstall

      mv deps/${old.pname}/dist $out
    
       runHook postInstall
      '';
    };
```

Secondly: you need to give `yarnOfflineCache = fetchYarnDeps { ... }` (refer to e.g: `tandoor-recipes` for example). I think `mkYarnPackage` will result in IFD otherwise, which is forbidden in nixpkgs.

Thirdly: what's happening with `node_module` in `buildPhase`?

---

I honestly cannot explain you what happens here, I just figured out a way to get my ReactJS / VueJS compilation working in Nix (as I didn't have any `main.js` output using the base setup), and this abomination was the result.
Maybe because I don't know JS ecosystem at all, just trying to build the apps :sweat_smile: 

I looked at the implementation done in `tandoor-recipes` and that seams to work well !